beta-src/src/components/ui/WDCountdownPill.tsx

```diff
@@ -39,26 +45,25 @@ const WDCountdownPill: React.FC<WDCountdownPillProps> = function ({
   const quarterTimeRemaining =
     endTimeInMilliSeconds - phaseTimeInMilliSeconds / 4;
 
-  const secondsLeft = endTime - +new Date() / milli;
-
-  const [timeLeft, setTimeLeft] = useState<ParsedTime>(
-    parseSeconds(secondsLeft),
+  const [formattedTimeLeft, setFormattedTimeLeft] = useState<string>(
+    getFormattedTimeLeft(endTime),
   );
 
   useEffect(() => {
-    const timer = setTimeout(() => {
-      setTimeLeft(parseSeconds(secondsLeft));
+    const timer = setInterval(() => {
+      const newFormattedTimeLeft = getFormattedTimeLeft(endTime);
+      setFormattedTimeLeft(newFormattedTimeLeft);
     }, 1000);
 
-    return () => clearTimeout(timer);
-  }, [secondsLeft]);
+    return () => clearInterval(timer);
+  }, []);
```

Comment:
I think you do want to include the endTime as a dependency here so useEffect runs again when the endTime changes. I'm assuming this is the turn end time, so for fast games could happen while the user is on screen. 

```
 return () => clearInterval(timer);
  }, [endTime]);
```

Reply:
the useEffect dep array is optional, 

- not specified -> the effect runs after each render.
- emtry arr `[]` -> runs once for the init render
- has a stateful value (ex.  `[endTime]` ) -> runs everytime the value of `endTime` changes
( The last pt does not ensure up to date props, thats on us to make sure our functions have the latest data )

Reply:
Thanks!